harness/determined/launch/autodeepspeed.py

```diff
@@ -0,0 +1,198 @@
+"""
+autodeepspeed.py is the launch layer for DeepSpeedTrial in Determined.
+
+It launches the entrypoint script using DeepSpeed's launch process.
+"""
+import argparse
+import logging
+import os
+import socket
+import subprocess
+import sys
+import time
+from typing import Dict, List, Tuple, cast
+
+from deepspeed.launcher.runner import DEEPSPEED_ENVIRONMENT_NAME
+
+import determined as det
+from determined import constants
+from determined.common import api
+from determined.common.api import certs
+
+
+def create_hostlist_file(num_proc_per_machine: int, ip_addresses: List[str]) -> Tuple:
+    trial_runner_hosts = ip_addresses.copy()
+    if len(ip_addresses) == 1:
+        trial_runner_hosts[0] = "localhost"
+    filename = "/tmp/hostfile.txt"
```

Comment:
DeepSpeed requires a hostfile and will default to single node dtrain without it.

I've changed it to use the default path that DeepSpeed checks for the hostfile.

Reply:
Can you pass it in as an argument instead?  Writing to `/job/hostfile` is much more obnoxious than `/tmp/hostfile.txt`.  This will also make it basically impossible to unit-test this code, because the unit test is going to have to run with like sudo privileges.

Reply:
i think i might have misunderstood your question.  are you asking whether the filename can get passed in as an argument or whether all the hosts can get passed in as an argument as in horovod?  the answer to the latter is no.  i will change this function to take the hostfile path as argument.